specification.md

```diff
@@ -145,10 +163,47 @@ configuration.
 See sections under the [Operation](#operation) section for the details of the
 message sequences.
 
+The WebSocket transport is typically used when it is necessary to have instant
+communication ability from the Server to the Agent without waiting for the Agent
+to poll the Server like it is done when using the HTTP transport (see below).
+
+## Plain HTTP Transport
+
+The second supported transport for OpAMP protocol is plain HTTP connection. The
+Agent is an HTTP client and the Server is an HTTP client server. The Agent makes
+POST requests to the Server. The body of the POST request and response is a
+[binary serialized Protobuf](https://developers.google.com/protocol-buffers/docs/encoding)
+message. The Agent sends AgentToServer Protobuf messages in the request body and
+the Server sends ServerToAgent Protobuf messages in the response body.
+
+OpAMP over HTTP is a synchronous, half-duplex message exchange protocol. The
+Agent initiates an HTTP request when it has an AgentToServer message to deliver.
+The Server responds to each HTTP request with a ServerToAgent message it wants
+to deliver to the Agent. If the Agent has nothing to deliver to the Server the
+Agent MUST periodically poll the Server by sending an AgentToServer message
+where only [instance_uid](#instance_uid) field is set. This gives the Server an
+opportunity to send back in the response any messages that the Server wants to
+deliver to the Agent (such as for example a new remote configuration).
+
+The default polling interval when the Agent does have anything to deliver is 30
+seconds. This polling interval SHOULD be configurable on the Agent.
+
+When using HTTP transport the sequence messages is exactly the same as it is
+when using the WebSocket transport. The only difference is in the timing in the
+situation when the Server wants to send a message to the Agent but the Server
+needs to wait for the Agent to poll the Server and establish an HTTP request
+over which the Server's message can be sent back.
```

Comment:
It took me a while to process the last sentence so just wanted to make sure I got it right. When Agent has something to send to the Server, it does not need to wait for the polling interval to kick, it can send the message anytime, for which the server might return another message (and the conversation might continue further). 

When Server has something to send to the Agent (e.g. ConnectioSettingsOffer), it might need to wait for the "ping" from the Agent.

Reply:
Correct. I probably need to rewrite this to make it clear.

Reply:
Made some clarifications.